### PR TITLE
Add ability to concede + fix GameRoom creation bug

### PR DESCRIPTION
**Pull request by zheng-ze**

### CODE REVIEW

File: TowerForge/TowerForge/GameModule/GameModes/DeathMatchMode.swift [hunk 1]
```diff
@@ -45,14 +45,18 @@ class DeathMatchMode: GameMode {
...
-                gameState = .WIN
-            } else if currentOwnKillCounter < currentOpponentKillCounter {
-                gameState = .LOSE
-            } else {
-                gameState = .DRAW
-            }
+        guard gameState != .LOSE else {
+            return
+        }
+        guard remainingTime < 0 else {
```
Comment (Vanessamae23) [inline TowerForge/TowerForge/GameModule/GameModes/DeathMatchMode.swift:51]: hmm why is this a `return`

File: TowerForge/TowerForge/GameModule/GameModes/SurvivalGameMode.swift [hunk 2]
```diff
@@ -54,6 +54,9 @@ class SurvivalGameMode: GameMode {
 
     func updateGame(deltaTime: TimeInterval) {
         nextWaveTime -= deltaTime
+        guard gameState != .LOSE else {
```
Comment (Vanessamae23) [inline TowerForge/TowerForge/GameModule/GameModes/SurvivalGameMode.swift:57]: what's the rationale of putting this guard statement?

**Review by Vanessamae23 — COMMENTED** [review]
just a few questions

File: TowerForge/TowerForge/GameModule/GameModes/DeathMatchMode.swift (hunk 1 above)
Comment (zheng-ze) [reply TowerForge/TowerForge/GameModule/GameModes/DeathMatchMode.swift:51]: So that we don't override the concede when updateGame is called.

**Review by Vanessamae23 — APPROVED** [review]
lgtm

**Pull request merged**